packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -131,25 +129,19 @@ class StorageS3Service {
   /// service returned [smithy.UnknownSmithyHttpException] if any.
   /// {@endtemplate}
   Future<S3ListResult> list({
-    String? path,
+    required StoragePath path,
     required StorageListOptions options,
   }) async {
     final s3PluginOptions = options.pluginOptions as S3ListPluginOptions? ??
         const S3ListPluginOptions();
-    final resolvedPrefix = await getResolvedPrefix(
-      prefixResolver: _prefixResolver,
-      logger: _logger,
-      accessLevel: options.accessLevel ?? _s3PluginConfig.defaultAccessLevel,
-      identityId: s3PluginOptions.targetIdentityId,
-    );
 
-    final listTargetPrefix = '$resolvedPrefix${path ?? ''}';
+    final fullPath = await _pathResolver.resolvePath(path: path);
```

Comment:
*nit:* for consistency let's name this resolvedPath
```suggestion
    final resolvedPath = await _pathResolver.resolvePath(path: path);
```